fastparquet/dataframe.py

```diff
@@ -94,10 +96,13 @@ def cat(col):
             df[six.text_type(col)] = Categorical([], categories=cat(col),
                                                  fastpath=True)
         else:
-            if hasattr(t, 'base'):
+            if hasattr(t, 'base') and t.base is not None:
                 # funky pandas not-dtype
                 t = t.base
-            d = np.empty(0, dtype=t)
+            if hasattr(t,'na_value'):
```

Comment:
`pd.api.extensions.is_extension_array_dtype(t)` is probably the best way to check this.

Reply:
I've used pd.api.types.is_extension_array_dtype(t)